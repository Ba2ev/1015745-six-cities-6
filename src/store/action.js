export const ActionType = {
  UPDATE_CITY: `city/updateCity`,
  UPDATE_SORT: `sort/updateSort`,
  TOGGLE_SORTLIST: `sort/toggleSortList`,
  UPDATE_HOVERED_OFFER_ID: `offer/updateHoveredOfferId`,
  LOAD_OFFERS: `offers/loadOffers`,
  LOAD_FAVORITES_OFFERS: `offers/loadFavoritesOffers`,
  UPDATE_FAVORITES_OFFERS: `offers/updateFavoritesOffers`,
  LOAD_OFFER_DATA: `offer/loadOfferData`,
  LOAD_OFFER_COMMENTS: `offer/loadOfferComments`,
  LOAD_OFFER_NEARBY: `offer/loadOfferNearby`,
  REQUIRED_AUTHORIZATION: `user/requiredAuthorization`,
  UPDATE_ACCOUNT: `user/updateAccount`,
};

export const ActionCreator = {
  updateCity: (city) => ({
    type: ActionType.UPDATE_CITY,
    payload: city
  }),
  updateSort: (sortType) => ({
    type: ActionType.UPDATE_SORT,
    payload: sortType
  }),
  toggleSortList: () => ({
    type: ActionType.TOGGLE_SORTLIST,
  }),
  updateHoveredOfferId: (id) => ({
    type: ActionType.UPDATE_HOVERED_OFFER_ID,
    payload: id
  }),
  loadOffers: (offers) => ({
    type: ActionType.LOAD_OFFERS,
    payload: offers
  }),
  loadFavoritesOffer: (offers) => ({
    type: ActionType.LOAD_FAVORITES_OFFERS,
    payload: offers
  }),
  loadOfferNearby: (nearblyOffers) => ({
    type: ActionType.LOAD_OFFER_NEARBY,
    payload: nearblyOffers
  }),
  loadOfferData: (offerData) => ({
    type: ActionType.LOAD_OFFER_DATA,
    payload: offerData
  }),
  loadOfferComments: (offerComments) => ({
    type: ActionType.LOAD_OFFER_COMMENTS,
    payload: offerComments
  }),
  updateFavoriteOffers: (offerData) => ({
    type: ActionType.UPDATE_FAVORITES_OFFERS,
    payload: offerData
  }),
  requireAuthorization: (status) => ({
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status,
  }),
  updateAccount: (email) => ({
    type: ActionType.UPDATE_ACCOUNT,
    payload: email,
  }),
};

