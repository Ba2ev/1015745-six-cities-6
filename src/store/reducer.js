import {combineReducers} from 'redux';
import {user} from './user/user';
import {property} from './property/property';
import {places} from './places/places';

export const NameSpace = {
  PROPERTY: `PROPERTY`,
  PLACES: `PLACES`,
  USER: `USER`
};

export default combineReducers({
  [NameSpace.PROPERTY]: property,
  [NameSpace.PLACES]: places,
  [NameSpace.USER]: user,
});
