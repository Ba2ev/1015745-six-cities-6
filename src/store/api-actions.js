import {loadOffers, loadOfferData, loadOfferComments, loadOfferNearby, loadFavoritesOffer, updateOfferFavorite, requireAuthorization, updateAccount} from './action';
import {adaptOfferToClient, adaptCommentToClient} from '../adapter/adapter';

export const fetchOffers = () => (dispatch, _getState, api) => (
  api.get(`/hotels`)
    .then(({data}) => (data.map(adaptOfferToClient)))
    .then((data) => dispatch(loadOffers(data)))
);

export const fetchOfferData = (id) => (dispatch, _getState, api) => (
  api.get(`/hotels/${id}`)
    .then(({data}) => (adaptOfferToClient(data)))
    .then((data) => dispatch(loadOfferData(data)))
);

export const fetchOfferComments = (id) => (dispatch, _getState, api) => (
  api.get(`/comments/${id}`)
    .then(({data}) => (data.map(adaptCommentToClient)))
    .then((data) => dispatch(loadOfferComments(data)))
);

export const fetchOfferNearby = (id) => (dispatch, _getState, api) => (
  api.get(`/hotels/${id}/nearby`)
    .then(({data}) => (data.map(adaptOfferToClient)))
    .then((data) => dispatch(loadOfferNearby(data)))
);

export const fetchFavoritesOffer = () => (dispatch, _getState, api) => (
  api.get(`/favorite`)
    .then(({data}) => (data.map(adaptOfferToClient)))
    .then((data) => dispatch(loadFavoritesOffer(data)))
);

export const changeFavorite = ({id, status}) => (dispatch, _getState, api) => (
  api.post(`/favorite/${id}/${status}`)
    .then(({data}) => (adaptOfferToClient(data)))
    .then((data) => dispatch(updateOfferFavorite(data)))
);

export const sendComment = ({id, comment, rating}) => (dispatch, _getState, api) => (
  api.post(`/comments/${id}`, {comment, rating})
    .then(({data}) => (data.map(adaptCommentToClient)))
    .then((data) => dispatch(loadOfferComments(data)))
);

export const checkAuth = () => (dispatch, _getState, api) => (
  api.get(`/login`)
    .then(({data}) => dispatch(updateAccount(data.email)))
    .then(() => dispatch(requireAuthorization(true)))
    .catch(() => {})
);

export const login = ({login: email, password}) => (dispatch, _getState, api) => (
  api.post(`/login`, {email, password})
    .then(() => dispatch(requireAuthorization(true)))
    .then(() => dispatch(updateAccount(email)))
);

export const logout = () => (dispatch, _getState, api) => (
  api.get(`/logout`)
    .then(() => dispatch(requireAuthorization(false)))
    .then(() => dispatch(updateAccount(``)))
);
