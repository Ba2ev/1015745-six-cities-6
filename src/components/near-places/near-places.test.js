import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import {createMemoryHistory} from 'history';
import NearPlaces from './near-places';
import {NameSpace} from '../../store/reducer';

const nearbyOffers = [
  {
    "city": {
      "name": `Paris`,
      "location": {
        "latitude": 48.85661,
        "longitude": 2.351499,
        "zoom": 13
      }
    },
    "previewImage": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    "images": [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`
    ],
    "title": `The Pondhouse - A Magical Place`,
    "isFavorite": false,
    "isPremium": false,
    "rating": 2.8,
    "type": `hotel`,
    "bedrooms": 3,
    "maxAdults": 6,
    "price": 295,
    "goods": [
      `Laptop friendly workspace`,
      `Fridge`,
      `Coffee machine`,
      `Baby seat`,
      `Dishwasher`,
      `Breakfast`,
      `Towels`,
      `Washer`,
      `Air conditioning`
    ],
    "host": {
      "id": 25,
      "name": `Angelina`,
      "isPro": true,
      "avatarUrl": `img/avatar-angelina.jpg`
    },
    "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    "location": {
      "latitude": 48.868610000000004,
      "longitude": 2.342499,
      "zoom": 16
    },
    "id": 103
  },
  {
    "city": {
      "name": `Paris`,
      "location": {
        "latitude": 48.85661,
        "longitude": 2.351499,
        "zoom": 13
      }
    },
    "previewImage": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
    "images": [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`
    ],
    "title": `Waterfront with extraordinary view`,
    "isFavorite": false,
    "isPremium": false,
    "rating": 4.4,
    "type": `room`,
    "bedrooms": 1,
    "maxAdults": 3,
    "price": 123,
    "goods": [
      `Baby seat`,
      `Washer`,
      `Laptop friendly workspace`,
      `Breakfast`,
      `Air conditioning`
    ],
    "host": {
      "id": 25,
      "name": `Angelina`,
      "isPro": true,
      "avatarUrl": `img/avatar-angelina.jpg`
    },
    "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    "location": {
      "latitude": 48.858610000000006,
      "longitude": 2.330499,
      "zoom": 16
    },
    "id": 102
  },
  {
    "city": {
      "name": `Paris`,
      "location": {
        "latitude": 48.85661,
        "longitude": 2.351499,
        "zoom": 13
      }
    },
    "previewImage": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    "images": [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`
    ],
    "title": `Wood and stone place`,
    "isFavorite": false,
    "isPremium": false,
    "rating": 3,
    "type": `house`,
    "bedrooms": 2,
    "maxAdults": 8,
    "price": 560,
    "goods": [
      `Washer`,
      `Dishwasher`,
      `Air conditioning`,
      `Breakfast`,
      `Fridge`,
      `Baby seat`,
      `Laptop friendly workspace`,
      `Towels`
    ],
    "host": {
      "id": 25,
      "name": `Angelina`,
      "isPro": true,
      "avatarUrl": `img/avatar-angelina.jpg`
    },
    "description": `A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.`,
    "location": {
      "latitude": 48.865610000000004,
      "longitude": 2.350499,
      "zoom": 16
    },
    "id": 33
  }
];

const mockStore = configureStore({});

it(`NearPlaces should render correctly`, () => {
  const history = createMemoryHistory();

  const initialState = {
    [NameSpace.USER]: {
      isAuthorized: true,
    }
  };

  const {getByText} = render(
      <redux.Provider store={mockStore(initialState)}>
        <Router history={history}>
          <NearPlaces offers={nearbyOffers}/>
        </Router>
      </redux.Provider>
  );

  const title = getByText(`Other places in the neighbourhood`);

  expect(title).toBeInTheDocument();
});
