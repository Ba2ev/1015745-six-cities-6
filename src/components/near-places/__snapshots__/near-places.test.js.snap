// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NearPlaces should render snapshot correctly 1`] = `
<div>
  <section
    class="near-places places"
  >
    <h2
      class="near-places__title"
    >
      Other places in the neighbourhood
    </h2>
    <div
      class="near-places__list places__list"
    >
      <article
        class="near-places__card place-card"
        data-cart-id="103"
      >
        <div
          class="near-places__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/103"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="200"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          class=" place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                295
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              class="place-card__bookmark-button  button"
              type="button"
            >
              <svg
                class="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 56%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/103"
            >
              The Pondhouse - A Magical Place
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            hotel
          </p>
        </div>
      </article>
      <article
        class="near-places__card place-card"
        data-cart-id="102"
      >
        <div
          class="near-places__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/102"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="200"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          class=" place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                123
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              class="place-card__bookmark-button  button"
              type="button"
            >
              <svg
                class="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 88%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/102"
            >
              Waterfront with extraordinary view
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            room
          </p>
        </div>
      </article>
      <article
        class="near-places__card place-card"
        data-cart-id="33"
      >
        <div
          class="near-places__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/33"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="200"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          class=" place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                560
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              class="place-card__bookmark-button  button"
              type="button"
            >
              <svg
                class="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 60%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/33"
            >
              Wood and stone place
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            house
          </p>
        </div>
      </article>
    </div>
  </section>
</div>
`;
